workers/data_refinery_workers/processors/illumina.R

```diff
@@ -424,8 +428,27 @@ probesToKeep <- intersect(probesToKeep, rownames(normData))
 normData <- normData[probesToKeep,,drop=FALSE]
 
 # Map probes to genes
-probeGene <- as.data.frame(probeGeneRef[mappedkeys(probeGeneRef)])
-rownames(probeGene) <- probeGene$IlluminaID
+probe_map <- read.csv(paste0("/home/user/probe_maps/", platform, ".tsv"), sep="\t", na.strings=c(""))
+rownames(probe_map) <- probe_map$probe_id
+
+choose_gene = function(probe_id, ensembl_ids) {
+  probe_id <- probe_id[1]
+  if (length(ensembl_ids) == 1) {
+    return(ensembl_ids[1])
+  }
+
+  # TODO: emit some kind of warning. Also, we need to decide what to do about NAs
+  probe_map[probe_id,"ensembl_id"][1]
```

Comment:
I agree that this is a little bit hacky. My Illumina refinery only returns the first preference, so there should only be one element, but R always returns a vector when you index this way so I am basically doing `c(1)[1]`. Is there a different way I can index so that I don't have to extract the one value in the vector like this?

Reply:
It turns out that the indexing was unnecessary, so I removed it to avoid confusion.